pkg/mls/api/v1/service.go

```diff
@@ -295,7 +295,8 @@ func (s *Service) SendWelcomeMessages(ctx context.Context, req *mlsv1.SendWelcom
 			insertSpan, insertCtx := tracer.StartSpanFromContext(ctx, "insert-welcome-message")
 			insertLogger := tracing.Link(insertSpan, log)
 			insertLogger.Info("inserting welcome message", zap.String("client_ip", ip), zap.Int("message_length", len(input.GetV1().Data)))
-			msg, err := s.store.InsertWelcomeMessage(insertCtx, input.GetV1().InstallationKey, input.GetV1().Data, input.GetV1().HpkePublicKey, types.WrapperAlgorithmFromProto(input.GetV1().WrapperAlgorithm))
+			msg, err := s.store.InsertWelcomeMessage(insertCtx, input.GetV1().InstallationKey, input.GetV1().Data, input.GetV1().HpkePublicKey, types.WrapperAlgorithmFromProto(input.GetV1().WrapperAlgorithm), int64(input.GetV1().MessageCursor))
```

Comment:
This is not due to your change, but if `input.GetV1()` returns nil, e.g. from a malformed or malicious payload, we get a nil dereference and this method will panic. I'm not sure what happens to the server in that case.

Might be nice to add a nil-check on the line before this one, or use the proto methods for fetching properties (e.g. `input.GetV1().GetInstallationKey()` instead of `input.GetV1().InstallationKey`)

Reply:
Done